src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResendNewPinCallbackHandler.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::resendNewPin
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse resendNewPin(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+
+        if (getDefendantLinkStatus(claim).isLinked()) {
+            logger.info("Claim {} has already been linked to defendant - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim has already been linked to defendant - cannot send notification"))
+                .build();
+        }
+
+        if (!claim.getClaimData().getDefendant().getEmail().isPresent()) {
+            logger.info("Claim {} doesn't have defendant email address - cannot send notification",
+                claim.getExternalId());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim doesn't have defendant email address - cannot send notification"))
+                .build();
+        }
+
+        GeneratePinResponse pinResponse = getPinResponse(claim.getClaimData(), authorisation)
+            .orElseThrow(() -> new IllegalArgumentException("Pin generation failed"));
+
+        claim.getClaimData().getDefendant().getEmail().ifPresent(defendantEmail ->
+            claimIssuedNotificationService.sendMail(
+                claim,
+                defendantEmail,
+                pinResponse.getPin(),
+                notificationsProperties.getTemplates().getEmail().getDefendantClaimIssued(),
+                "defendant-issue-notification-" + claim.getReferenceNumber(),
+                claim.getClaimData().getDefendant().getName()
+            ));
+
+        Claim updatedClaim = claim.toBuilder()
+            .letterHolderId(pinResponse.getUserId())
+            .build();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetailsConverter.convertToMap(caseMapper.to(updatedClaim)))
+            .build();
+    }
+
+    private Optional<GeneratePinResponse> getPinResponse(ClaimData claimData, String authorisation) {
+        return Optional.of(userService.generatePin(claimData.getDefendant().getName(), authorisation));
+    }
+
+    private DefendantLinkStatus getDefendantLinkStatus(Claim claim) {
+        if (claim.getDefendantId() == null) {
+            return new DefendantLinkStatus(false);
+        }
+        return new DefendantLinkStatus(true);
+    }
```

Comment:
Can be simplified to
```
return new DefendantLinkStatus(claim.getDefendantId() != null);
```

Though I would probably just change the if statement from 
```
if(getDefendantLinkStatus(claim).isLinked()){
```

to 
```
if(claim.getDefendantId() != null)){
```

and remove this method

Reply:
agree, done.